components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -187,11 +187,11 @@ public Response addCommentToAPI(String apiId, CommentDTO body, MessageContext me
         String requestedTenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
         try {
             APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
-            APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifierFromUUID(apiId, requestedTenantDomain);
+            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(apiId, requestedTenantDomain);
 
             Comment comment = CommentMappingUtil.fromDTOToComment(body, username, apiId);
-            String createdCommentId = apiConsumer.addComment(apiIdentifier, comment, username);
-            Comment createdComment = apiConsumer.getComment(apiIdentifier, createdCommentId);
+            String createdCommentId = apiConsumer.addComment(apiTypeWrapper, comment, username);
```

Comment:
We need to check whether the comment belongs to API or API_Product when deciding the SQL query. Since we need that information in DAO class, `ApiTypeWrapper` would be helpful as it has method - `isAPIProduct()`.

Reply:
For that case, we can use `if (identifier instanceof APIProductIdentifier)`

Reply:
Fixed. thanks :)